src/doc/DocHighlightThread.js

```diff
@@ -333,52 +331,84 @@ class DocHighlightThread extends AnnotationThread {
     setupElement() {}
 
     /**
-     * Binds custom event listeners for the dialog.
+     * Clear text selection and show annotation dialog on 'annotationdraw'
      *
-     * @protected
-     * @override
+     * @private
      * @return {void}
      */
-    /* istanbul ignore next */
-    bindCustomListenersOnDialog() {
-        // Annotation drawn
-        this.dialog.addListener('annotationdraw', () => {
-            this.state = STATES.pending_active;
-            window.getSelection().removeAllRanges();
-            this.show();
-        });
+    handleDraw() {
+        this.state = STATES.pending_active;
+        window.getSelection().removeAllRanges();
+        this.show();
+    }
 
-        // Annotation drawn
-        this.dialog.addListener('annotationcommentpending', () => {
-            this.state = STATES.pending_active;
-        });
+    /**
+     * Set the thread state to pending active on 'annotationcommentpending'
+     *
+     * @private
+     * @return {void}
+     */
+    handleCommentPending() {
+        this.state = STATES.pending_active;
+    }
 
-        // Annotation created
-        this.dialog.addListener('annotationcreate', (data) => {
-            if (data) {
-                this.type = TYPES.highlight_comment;
-                this.dialog.toggleHighlightCommentsReply(Object.keys(this.annotations).length);
-            } else {
-                this.type = TYPES.highlight;
-            }
+    /**
+     * Create the appropriate type of highlight annotation thread on
+     * 'annotationcreate'
+     *
+     * @private
+     * @param {Object} data Event data
+     * @return {void}
+     */
+    handleCreate(data) {
+        if (data) {
+            this.type = TYPES.highlight_comment;
+            this.dialog.toggleHighlightCommentsReply(Object.keys(this.annotations).length);
+        } else {
+            this.type = TYPES.highlight;
+        }
 
-            this.saveAnnotation(this.type, data ? data.text : '');
-        });
+        this.saveAnnotation(this.type, data ? data.text : '');
+    }
 
-        // Annotation canceled
-        this.dialog.addListener('annotationcancel', () => {
-            this.cancelFirstComment();
-        });
+    /**
+     * Delete the annotation annotation or the thread's first annotation based on
+     * if an annotationID is specified on 'annotationdelete'
+     *
+     * @private
+     * @param {Object} data Event data
+     * @return {void}
+     */
+    handleDelete(data) {
+        const firstAnnotation = annotatorUtil.getFirstAnnotation(this.annotations);
```

Comment:
Can you move `firstAnnotation` assignment to inside the `else if` where it's being used?